### DISCUSSION TITLE
pages build and deployment ERROR

### DISCUSSION THREAD

**Discussion by NirajPudasaini** [alshedivat/al-folio#2907] [Q&A]
Run actions/jekyll-build-pages@v1
/usr/bin/docker run --name ghcrioactionsjekyllbuildpagesv1013_6[9](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:10)a573 --label c689c5 --workdir /github/workspace --rm -e "INPUT_SOURCE" -e "INPUT_DESTINATION" -e "INPUT_FUTURE" -e "INPUT_BUILD_REVISION" -e "INPUT_VERBOSE" -e "INPUT_TOKEN" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_ID_TOKEN_REQUEST_URL" -e "ACTIONS_ID_TOKEN_REQUEST_TOKEN" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/nirajpudasaini.com.np/nirajpudasaini.com.np":"/github/workspace" ghcr.io/actions/jekyll-build-pages:v1.0.13
The following gems are missing
 * classifier-reborn (2.3.0)
 * jekyll (4.3.4)
 * jekyll-archives (2.2.1)
 * jekyll-email-protect (1.1.0)
 * jekyll-get-json (1.0.0)
 * jekyll-imagemagick (1.4.0)
 * jekyll-jupyter-notebook (0.0.6)
 * jekyll-link-attributes (1.0.1)
 * jekyll-minifier (0.1.[10](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:11))
 * jekyll-paginate-v2 (3.0.0)
 * jekyll-regex-replace (1.1.0)
 * jekyll-scholar (7.1.3)
 * jekyll-tabs (1.2.1)
 * jekyll-toc (0.19.0)
 * jekyll-twitter-plugin (2.1.0)
 * unicode_utils (1.4.0)
 * webrick (1.9.1)
 * css_parser (1.17.1)
 * feedjira (3.2.3)
 * httparty (0.22.0)
 * fast-stemmer (1.0.2)
 * i18n (1.14.6)
 * jekyll-sass-converter (3.0.0)
 * kramdown (2.5.1)
 * mercenary (0.4.0)
 * rouge (4.5.1)
 * terminal-table (3.0.2)
 * deep_merge (1.2.2)
 * cssminify2 (2.0.1)
 * htmlcompressor (0.4.0)
 * json-minify (0.0.3)
 * uglifier (4.2.0)
 * bibtex-ruby (6.1.0)
 * citeproc-ruby (1.1.14)
 * csl-styles (1.0.1.[11](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:12))
 * nokogiri (1.16.6)
 * loofah (2.22.0)
 * sax-machine (1.3.2)
 * mini_mime (1.1.5)
 * multi_xml (0.7.1)
 * public_suffix (6.0.1)
 * concurrent-ruby (1.3.4)
 * sass-embedded (1.83.0)
 * rexml (3.4.0)
 * unicode-display_width (2.6.0)
 * json (2.7.2)
 * latex-decode (0.4.0)
 * racc (1.8.0)
 * citeproc (1.0.10)
 * csl (1.6.0)
 * crass (1.0.6)
 * google-protobuf (4.29.2)
 * namae (1.2.0)
 * rake (13.2.1)
Install missing gems with `bundle install`
Warning: The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/
To use retry middleware with Faraday v2.0+, install `faraday-retry` gem
  Liquid Exception: Liquid syntax error (line 64): Unknown tag 'toc' in /_layouts/post.liquid
/usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block.rb:38:in `unknown_tag': Liquid syntax error (line 64): Unknown tag 'toc' (Liquid::SyntaxError)
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:38:in `unknown_tag'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block.rb:68:in `block in parse_body'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:32:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block.rb:58:in `parse_body'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tags/if.rb:30:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/tag.rb:10:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/block_body.rb:34:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:10:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/document.rb:5:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:130:in `parse'
	from /usr/local/bundle/gems/liquid-4.0.4/lib/liquid/template.rb:114:in `parse'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:13:in `block in parse'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:49:in `measure_time'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/liquid_renderer/file.rb:[12](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:13):in `parse'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:121:in `render_liquid'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:195:in `render_layout'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:[15](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:16)8:in `place_in_layouts'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:88:in `render_document'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/renderer.rb:62:in `run'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:479:in `render_regenerated'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:464:in `block (2 levels) in render_docs'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:463:in `each'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:463:in `block in render_docs'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:462:in `each_value'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:462:in `render_docs'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:191:in `render'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/site.rb:71:in `process'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/command.rb:28:in `process_site'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/commands/build.rb:65:in `build'
	from /usr/local/bundle/gems/jekyll-3.10.0/lib/jekyll/commands/build.rb:36:in `process'
	from /usr/local/bundle/gems/github-pages-232/bin/github-pages:70:in `block (3 levels) in <top (required)>'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /usr/local/bundle/gems/github-pages-232/bin/github-pages:6:in `<top (required)>'
	from /usr/local/bundle/bin/github-pages:25:in `load'
	from /usr/local/bundle/bin/github-pages:25:in `<main>'
Error:  Logging at level: debug Configuration file: /github/workspace/./_config.yml GitHub Pages: github-pages v232 GitHub Pages: jekyll v3.10.0 Theme: jekyll-theme-primer Theme source: /usr/local/bundle/gems/jekyll-theme-primer-0.6.0 Requiring: jekyll-github-metadata Requiring: jekyll-seo-tag Requiring: jekyll-feed Requiring: jekyll-sitemap Requiring: jemoji Requiring: jekyll-coffeescript Requiring: jekyll-commonmark-ghpages Requiring: jekyll-gist Requiring: jekyll-github-metadata Requiring: jekyll-paginate Requiring: jekyll-relative-links Requiring: jekyll-optional-front-matter Requiring: jekyll-readme-index Requiring: jekyll-default-layout Requiring: jekyll-titles-from-headings GitHub Metadata: Initializing... Source: /github/workspace/. Destination: /github/workspace/./_site Incremental build: disabled. Enable with --incremental Generating... EntryFilter: excluded /purgecss.config.js EntryFilter: excluded /INSTALL.md EntryFilter: excluded /CUSTOMIZE.md EntryFilter: excluded /CONTRIBUTING.md EntryFilter: excluded /docker-compose-slim.yml EntryFilter: excluded /Gemfile.lock EntryFilter: excluded /Gemfile EntryFilter: excluded /README.md EntryFilter: excluded /package-lock.json EntryFilter: excluded /Dockerfile EntryFilter: excluded /package.json EntryFilter: excluded /FAQ.md EntryFilter: excluded /LICENSE EntryFilter: excluded /docker-compose.yml Reading: _posts/2024-07-29-research_keywords.md Reading: _posts/2024-12-[18](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:19)-cnn-from-scratch.md EntryFilter: excluded /readme_preview/photos-screenshot.png EntryFilter: excluded /readme_preview/jupyter.png EntryFilter: excluded /readme_preview/al-folio-preview.png EntryFilter: excluded /readme_preview/distill.png EntryFilter: excluded /readme_preview/code.png EntryFilter: excluded /readme_preview/publications.png EntryFilter: excluded /readme_preview/cv.png EntryFilter: excluded /readme_preview/math.png EntryFilter: excluded /readme_preview/projects.png EntryFilter: excluded /readme_preview/dark.png EntryFilter: excluded /readme_preview/blog.png EntryFilter: excluded /readme_preview/light.png EntryFilter: excluded /readme_preview/people.png EntryFilter: excluded /readme_preview/repositories.png EntryFilter: excluded /_pages/about_einstein.md EntryFilter: excluded /bin/cibuild EntryFilter: excluded /bin/deploy EntryFilter: excluded /bin/entry_point.sh EntryFilter: excluded /lighthouse_results/mobile EntryFilter: excluded /lighthouse_results/desktop Reading: _news/announcement_1.md Reading: _news/announcement_2.md Reading: _news/announcement_3.md Reading: _news/announcement_4.md Reading: _news/announcement_5..md Reading: _news/announcement_6.md Reading: _projects/1_project.md Reading: _projects/2_project.md Reading: _projects/3_project.md Reading: _projects/4_project.md Reading: _projects/5_project.md Reading: _projects/6_project.md Reading: _projects/7_project.md Reading: _projects/8_project.md Reading: _projects/9_project.md Generating: JekyllOptionalFrontMatter::Generator finished in 0.000100808 seconds. Generating: JekyllReadmeIndex::Generator finished in 8.9156e-05 seconds. Jekyll Feed: Generating feed for posts Generating: JekyllFeed::Generator finished in 0.000795553 seconds. Generating: Jekyll::JekyllSitemap finished in 0.003073207 seconds. Generating: Jekyll::Paginate::Pagination finished in 1.0469e-05 seconds. Generating: JekyllRelativeLinks::Generator finished in 0.00504422 seconds. Generating: JekyllDefaultLayout::Generator finished in 0.000352527 seconds. Generating: JekyllTitlesFromHeadings::Generator finished in 2.1[19](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:20)9e-05 seconds. Rendering: _posts/[20](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:21)24-07-29-research_keywords.md Pre-Render Hooks: _posts/2024-07-29-research_keywords.md Rendering Markup: _posts/2024-07-29-research_keywords.md Requiring: kramdown-parser-gfm Rendering Layout: _posts/2024-07-29-research_keywords.md Layout source: site github-pages [23](https://github.com/NirajPudasaini/nirajpudasaini.com.np/actions/runs/12452385715/job/34761275687#step:4:24)2 | Error: Liquid syntax error (line 64): Unknown tag 'toc'

**Comment by george-gca** [comment]
Have you followed our [video tutorial](https://github.com/alshedivat/al-folio/blob/main/INSTALL.md#recommended-approach)?

**Reply by NirajPudasaini** [reply]
Hey, thanks for this. 
Note. I used gh-pages to publish and it worked. Thanks!
